### PR TITLE
Update License

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Updates the license for v3 to use the Six Labors Split License #2151

### CODE REVIEW

File: APACHE_LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
```
Comment (gfoidl) [inline APACHE_LICENSE:182]: There are no fields enclosed in `[ ]`.

Comment (JimBobSquarePants) [reply APACHE_LICENSE:182]: Ah good spot. I need to remove the instructions. 

I also need to update the copyright in the files!

**Comment by JimBobSquarePants** [comment]
I'm not going to update the license at the top of each file until someone has a chance to review these changes because it's too much noise.

After the review I will merge this PR https://github.com/SixLabors/SharedInfrastructure/pull/31 which will update the style config to use our new license. Then I'll do a simple find/replace to update each file before merging this PR.

**Comment by JimBobSquarePants** [comment]
@antonfirsov Seeing a new Windows issue here. 

```
 Xunit.Sdk.TrueException: leakCounter did not count enough leaks (2 only)
Expected: True
Actual:   False
   at SixLabors.ImageSharp.Tests.Memory.Allocators.MemoryDiagnosticsTests.<MissingCleanup_LeaksAreReported>g__RunTest|4_0(String isGroupInner, String subscribeLeakHandleInner) in D:\a\ImageSharp\ImageSharp\tests\ImageSharp.Tests\Memory\Allocators\MemoryDiagnosticsTests.cs:line 0
```

```
SixLabors.ImageSharp.Tests: BenchmarkDotNet=v0.13.0, OS=Windows 10.0.20348
Intel Xeon Platinum 8370C CPU 2.80GHz, 1 CPU, 2 logical and 2 physical cores
.NET SDK=7.0.100-preview.6.22352.1
  [Host] : .NET 6.0.7 (6.0.722.32202), X64 RyuJIT
```

I'm gonna split out the SDK installers to completely isolate the preview SDK.

**Comment by JimBobSquarePants** [comment]
I'm gonna merge this. I've updated everything to point at the new license and supplementary docs too.

**Comment by VisualMelon** [comment]
Sorry to ask a boring question, but wanted to ask for clarification on the point of OSS dependents which don't have restrictions on commercial use (e.g. MIT).  In OxyPlot, we may update to ImageSharp 2.1.3 and ImageSharp.Drawing 1.0.0-beta15, and I'd like to be sure that we're not missing anything with regard to licensing now or going forward (current diff under consideration: https://github.com/oxyplot/oxyplot/pull/1934/files#diff-cf34a87e1e75e04322c9c3dacb6efdc814544befbfbf56567f144835d394aa92 ). My reading of the recent blog post is that the situation after version 3 will be the same as version 2, and from my reading I think the situation is that we wouldn't need to do anything for 2.0 (per the first bullet in https://github.com/SixLabors/ImageSharp/blob/main/LICENSE and the nature of nuget), but I would like to sure on this matter.

**Comment by mistial-dev** [comment]
How does this work, practically speaking?

Suppose large company XYZ wants to use ImageSharp, and not pay for it.  

> Works in Source or Object form are licensed to You under the Apache License, Version 2.0 if.
> You are consuming the Work in for use in software licensed under an Open Source or Source Available license.

Company XYZ integrates ImageSharp into XYZImageSharp, an open source slightly extended version of ImageSharp, with the new code under the Apache 2 license.

That's "for use in software licensed under an Open Source license."  Their revenue doesn't matter anymore.

**Comment by JimBobSquarePants** [comment]
@mistial-dev 

See the following discussion. 

https://github.com/SixLabors/ImageSharp/discussions/2151#discussioncomment-2942963:~:text=determine%20liability%20(none).-,Can%20it%20be%20exploited.%20Yes%2C%20but%20as%20I%27ve%20said%20elsewhere.%20%22Evil%20Corp%20is%20gonna%20evil%22.%20They%27re%20actually%20more%20likely%20to%20simply%20ignore%20the%20license.,-Most%20won%27t%20blink

**Pull request merged**